src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -127,6 +127,46 @@ public async Task<IActionResult> GetRecommendations(string sessionId)
             return Ok(output);
         }
 
+        /// <summary>
+        /// Sets a value for a list of option setting items on the selected recommendation.
+        /// Option setting updates are provided as Key Value pairs with the Key being the JSON path to the leaf node.
+        /// Only primitive data types are supported for Value updates. The Value is a string value which will be parsed as its corresponding data type.
+        /// </summary>
+        [HttpPut("session/<sessionId>/settings")]
+        [SwaggerOperation(OperationId = "SetConfigSettings")]
+        [SwaggerResponse(200, type: typeof(SetConfigSettingsOutput))]
+        [Authorize]
+        public IActionResult SetConfigSettings(string sessionId, [FromBody] SetConfigSettingsInput input)
```

Comment:
nit: I kind of prefer the term `ApplyConfigSettings` instead of `SetConfigSettings` because it gives more indication there will be side effects when setting a specific setting.

Reply:
I changed the name.